### ISSUE TITLE
Haskell grammar parse error

### ISSUE THREAD

**Issue by sevketcakir** [antlr/grammars-v4#1685]
Hello,
I'm trying to parse the simple Haskell code below with the grammar but I'm getting the error below. Is it me doing something wrong?
code:

```hs
module Test where

amethod::[a]->[a]
amethod x = x
```

error:

```
line 0:-1 no viable alternative at input 'amethodSEMI'
line 3:7 mismatched input '::' expecting {'{-#', '(', '[', '-', '~', ';', NEWLINE, '}', 'VCCURLY', 'SEMI', 'class', 'data', 'default', 'foreign', 'infix', 'infixl', 'instance', 'newtype', 'type', '_', 'as', 'hiding', CHAR, STRING, VARID, CONID, DECIMAL, OCTAL, HEXADECIMAL, FLOAT}
line 4:0 extraneous input 'amethod' expecting {<EOF>, ';', 'SEMI'}
line 4:7 extraneous input 'VCCURLY' expecting {<EOF>, ';', 'SEMI'}
```

**Comment by sloboegen98** [comment]
Hi! This is bug with lexer. 
This code works fine:
```
module Test where

amethod :: [a]->[a]
amethod x = x
```

I will try to fix this bug.

**Comment by Octogonapus** [comment]
@teverett This issue should be closed now that #1695 is merged.
